src/controllers/ProductsController.js

```diff
@@ -762,6 +774,23 @@ class ProductsController extends Controller {
     output.id = product_id.toString();
     return output;
   }
+
+  _findAndUnreleaseCurrentFirmware(
+    productFirmwareList: Array<ProductFirmware>,
+  ): Promise<*> {
+    return Promise.all(
+      productFirmwareList
+        .filter(
```

Comment:
basically this should return 0 or 1 element, so I could use `find` instead, but `filter` seems a little bit safe